### ISSUE TITLE
increase granularity of dates in frequency graph

### ISSUE THREAD

**Issue by mduering** [impresso/impresso-frontend#91]
see screenshot below, it would be great if we could add months/ days depending on the zoom level
![image](https://user-images.githubusercontent.com/4408050/47950132-4c93d500-df4e-11e8-89f3-0b1446e918b0.png)

**Comment by tvanbeek** [comment]
At the moment the linechart data is a list of hits (y axis) per year (x axis). The chart supports more granular ticks for the date dimension (months/weeks/days) but we should figure out if this makes sense. Maybe we could add a dropdown in the collapsable filter settings so the user can switch between 'grouping per year/month/day' @danieleguido? Because this has to be supported by the middle layer.

**Comment by danieleguido** [comment]
Well, at the moment we only include the `year` facets along with the search query, with a limit of 500 buckets (that is, 500 group of results by year). 
There is the possibility of getting the `day` facet, where SOLR counts number of hits per day. 
The result is a list of the "top x" days though, which is totally useless in a timeline and we cannot raise the limit above 1000 for pref reason.
I won't let the users choose the granularity, as the facets is calculated no matter how many results retrieved and could lead to memory leak...

Instead I suggest that whenever the date range (min, max) of the results is in env. 2 years range, the middle layer automatically add a day facet that can be used to fill timeline data.

Basically the middle layer would perform two queries against SOLR in sequence: the first step is to get the list of results along with the min and max `day` of the results retrieved; then, if the date range is less than 500 days, the ML asks for the `day` facets, with a reasonable limit of 500.

This would be slower though, but not that much (100 ms more?): it depends on the list of results.

Does this make sense?

**Comment by danieleguido** [comment]
tbd in impresso-datalab :)
